distributed/scheduler.py

```diff
@@ -7052,6 +7083,8 @@ def get_metadata(self, keys: list[str], default=no_default):
     def set_restrictions(self, worker: dict[str, Collection[str] | str]):
         for key, restrictions in worker.items():
             ts = self.tasks[key]
+            if ts._rootish is not None:
```

Comment:
I don't like that `set_restrictions` is a public API at all. Doesn't seem like something you should be able to do post-hoc.

Reply:
This fails `test_reschedule_concurrent_requests_deadlock`, which sets restrictions on a processing task. https://github.com/dask/distributed/blob/711997e02f81493e1cba6a81123f07d4a059cc10/distributed/tests/test_steal.py#L1292-L1355

Reply:
> I don't like that set_restrictions is a public API at all. Doesn't seem like something you should be able to do post-hoc.

We can change it. First step is a deprecation warning